Applications/spire/source/book_view_ui_tester/book_view_test_controller_window.cpp

```diff
@@ -30,6 +30,44 @@ BookViewTestControllerWindow::BookViewTestControllerWindow(
   connect(m_model_update_period_spin_box, &QSpinBox::editingFinished,
     this, &BookViewTestControllerWindow::on_model_period_updated);
   layout->addWidget(m_model_update_period_spin_box, 1, 1);
+  auto line = new QFrame(this);
+  line->setFrameShape(QFrame::HLine);
+  layout->addWidget(line, 2, 1, 1, 2);
+  auto market_combo_box_label = new QLabel("Market:", this);
+  layout->addWidget(market_combo_box_label, 3, 0);
+  m_market_combo_box = new QComboBox(this);
+  auto markets = GetDefaultMarketDatabase().GetEntries();
+  for(auto market : markets) {
```

Comment:
Don't need to copy markets, instead do this:

<code>
  for(auto& market : GetDefaultMarketDatabase().GetEntries()) {
    ..
  }
</code>